alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultAuthorizationUtility.java

```diff
@@ -0,0 +1,136 @@
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.synopsys.integration.alert.common.enumeration.AccessOperation;
+import com.synopsys.integration.alert.common.persistence.accessor.AuthorizationUtil;
+import com.synopsys.integration.alert.common.persistence.model.PermissionMatrixModel;
+import com.synopsys.integration.alert.common.persistence.model.UserRoleModel;
+import com.synopsys.integration.alert.database.authorization.AccessOperationEntity;
+import com.synopsys.integration.alert.database.authorization.AccessOperationRepository;
+import com.synopsys.integration.alert.database.authorization.PermissionKeyEntity;
+import com.synopsys.integration.alert.database.authorization.PermissionKeyRepository;
+import com.synopsys.integration.alert.database.authorization.PermissionMatrixRelation;
+import com.synopsys.integration.alert.database.authorization.PermissionMatrixRepository;
+import com.synopsys.integration.alert.database.user.RoleEntity;
+import com.synopsys.integration.alert.database.user.RoleRepository;
+import com.synopsys.integration.alert.database.user.UserRoleRelation;
+import com.synopsys.integration.alert.database.user.UserRoleRepository;
+
+@Component
+@Transactional
+public class DefaultAuthorizationUtility implements AuthorizationUtil {
+    private final RoleRepository roleRepository;
+    private final UserRoleRepository userRoleRepository;
+    private final PermissionMatrixRepository permissionMatrixRepository;
+    private final AccessOperationRepository accessOperationRepository;
+    private final PermissionKeyRepository permissionKeyRepository;
+
+    @Autowired
+    public DefaultAuthorizationUtility(final RoleRepository roleRepository, final UserRoleRepository userRoleRepository, final PermissionMatrixRepository permissionMatrixRepository, final AccessOperationRepository accessOperationRepository,
+        final PermissionKeyRepository permissionKeyRepository) {
+        this.roleRepository = roleRepository;
+        this.userRoleRepository = userRoleRepository;
+        this.permissionMatrixRepository = permissionMatrixRepository;
+        this.accessOperationRepository = accessOperationRepository;
+        this.permissionKeyRepository = permissionKeyRepository;
+    }
+
+    @Override
+    public Set<UserRoleModel> createRoleModels(final Collection<Long> roleIds) {
+        final Set<UserRoleModel> userRoles = new LinkedHashSet<>();
+        for (final Long roleId : roleIds) {
+            final Optional<String> roleName = getRoleName(roleId);
+            roleName.ifPresent(role -> userRoles.add(UserRoleModel.of(role, readPermissionsForRole(roleId))));
```

Comment:
can we combine lines 78 and 79 so that we getRoleName then immediately check ifPresent?